Modules/TopGear/TopGear.lua

```diff
@@ -183,20 +183,31 @@ function CraftSim.TOPGEAR:GetProfessionGearFromInventory(recipeData)
 
         for bag = BANK_CONTAINER, NUM_BAG_SLOTS + NUM_BANKBAGSLOTS do
             for slot = 1, C_Container.GetContainerNumSlots(bag) do
-                local itemLink = C_Container.GetContainerItemLink(bag, slot)
-                if itemLink ~= nil then
-                    local itemSubType = select(3, C_Item.GetItemInfoInstant(itemLink))
-                    if itemSubType == currentProfession then
-                        local professionGear = CraftSim.ProfessionGear()
-                        professionGear:SetItem(itemLink)
-                        table.insert(inventoryGear, professionGear)
-
-                        if not GUTIL:Find(CraftSimRecipeDataCache.professionGearCache[crafterUID]
-                                [recipeData.professionData.professionInfo.profession]
-                                .availableProfessionGear, function(g) return g.itemLink == itemLink end) then
-                            table.insert(CraftSimRecipeDataCache.professionGearCache[crafterUID]
-                                [recipeData.professionData.professionInfo.profession]
-                                .availableProfessionGear, professionGear:Serialize())
+                local itemLoc = ItemLocation:CreateFromBagAndSlot(bag,slot)
+                if itemLoc:IsValid() then
+                    local isCached = C_Item.IsItemDataCached(itemLoc)
+                    if not isCached then
+                        C_Item.RequestLoadItemData(itemLoc)

```

Comment:
afaik when you somehow trigger a query for an item's info it starts loading (e.g. GetItemInfo)
I never had the case where I needed to request it specifically

Reply:
I replied on discord for that, about the C_Item instead of C_Container